ext/ldap/ldap.c

```diff
@@ -424,6 +423,10 @@ static int _php_ldap_control_from_array(LDAP *ld, LDAPControl** ctrl, zval* arra
 							goto failure;
 						}
 
+						if (tmpstring != NULL) {
+							zend_string_release(tmpstring);
+						}
```

Comment:
This doesn't look right. It will free the string from the previous iteration, but a reference to it is stored inside ldap_attrs[i-1] :/

Reply:
Potentialy better solution in 370e7659b9ad202177e4488e6c15d5e34287cf7b